packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -56,10 +56,32 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   [registrar addMethodCallDelegate:instance channel:channel];
 }
 
+/**
+ * Initializes the _imagePickerController member with a new instance of the
+ * UIImagePickerController class.
+ *
+ * A new instance of the UIImagePickerController is created every time the
+ * initImagePickerController method is called. For testing purposes this can
+ * be overriden using the setImagePickerControllerOverride method, in which
+ * case the set instance of the UIImagePickerController is used to initialize
+ * the _imagePickerController member.
+ */
+- (void)initImagePickerController {
```

Comment:
`init*` is a confusing name for a method in ObjC that's not a class initializer. I think the structure of having this have a side-effect also makes the code at the call side harder to follow (and it locks us into having the controller as an ivar, which I'm not convinced is actually a good idea; it seems like we could make this less plugin stateful and harder to order incorrectly by passing it around).

Let's use [this pattern](https://github.com/flutter/plugins/blob/main/packages/local_auth/local_auth/ios/Classes/FLTLocalAuthPlugin.m#L49-L60) instead, which keeps the assignment as it was before, and just replaces the creation with a method call.